src/UnitTests/Analysis/StackPointerBackpropagatorTests.cs

```diff
@@ -0,0 +1,204 @@
+#region License
+/* 
+ * Copyright (C) 1999-2018 John Källén.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using NUnit.Framework;
+using Reko.Analysis;
+using Reko.Core;
+using Reko.UnitTests.Mocks;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Reko.UnitTests.Analysis
+{
+    [TestFixture]
+    public class StackPointerBackpropagatorTests
+    {
+        private Program program;
+
+        [SetUp]
+        public void Setup()
+        {
+            var arch = new FakeArchitecture();
+            var platform = new FakePlatform(null, arch);
+            this.program = new Program
+            {
+                Architecture = arch,
+                Platform  = platform,
+            };
+        }
+
+        private void AssertStringsEqual(string sExp, SsaState ssa)
+        {
+            var sw = new StringWriter();
+            ssa.Procedure.Write(false, sw);
+            var sActual = sw.ToString();
+            if (sExp != sActual)
+            {
+                Debug.Print("{0}", sActual);
+                Assert.AreEqual(sExp, sActual);
+            }
+        }
+
+        private SsaState RunTest(ProcedureBuilder m)
+        {
+            var proc = m.Procedure;
+            var scc = new HashSet<Procedure> { proc };
+            var sst = new SsaTransform(program, m.Procedure, scc, null, null);
+            sst.Transform();
+            sst.AddUsesToExitBlock();
```

Comment:
I suggest to use `SsaProcedureBuilder` instead of `SsaTransform` running. The dependency between `SsaTransform` and `StackPointerBackpropagator` tests is not good. These are `StackPointerBackpropagator` tests, not `SsaTransform` tests. If we want test integration between `SsaTransform` and `StackPointerBackpropagator` we need create `DataFlowAnalysis` tests

Reply:
I'll look into fixing this later tonight.